Feature/do 1598 user manual (#103)
Cofinity-X/cofinity-x-ba-traceability-foss: docs/src/docs/user/user-manual.adoc

==== Admin

With administrator authorization, the user is primarily able to configure the technical configuration of the EDC connection. From a technical perspective, the admin only has read authorization for existing warnings, investigations and products.

== Navigation

Navigation is done based on the top menu.

image::../../images/arc42/user-guide/tx-navigation.png[Navigation]

=== Dashboard

Provides a fast overview about the amount of manufactured (my parts) and supplier parts (shared parts) and batches, as well as the amount of open investigations and alerts.
Lists the five most recent quality investigations and alerts to get an overview of the current state.

=== My Parts

Navigates to the own manufactured parts and batches list view.

=== Shared parts

Navigates to the supplier and customer parts and batches list view.

=== Quality investigation

Navigates to the inbox and outgoing investigations.

=== Quality alert

Navigates to the incoming and outgoing alerts.

=== Burger menu

Use the burger menu to use view user details, access the user manual, logout or change the language.

Supported languages:

> *English
> *German

Reviewer: **Fix:**

Leave a space between asterisk and list item.

```suggestion
* English
* German
```

Parts that are in a quality alert are highlighted yellow.

=== AsPlanned Lifecycle Parts

List view of own Parts with AsPlanned Lifecycle.
This includes data based on the aspect models of Use Case Traceability: AsPlanned, SerialPart, Batch.
Parts that are in a quality alert are highlighted yellow.

=== Parts selection -> Create Quality alert

Select one or multiple child components/parts/batches that are build into your part.
Selection will enable you to create a quality alert (notification) to your customers.
The quality alert will be added to a queue (queued & sent inbox) and not directly sent to the customers.

Once the quality alert is created you will get a pop-up and can directly navigate to the inbox for further action.

=== Parts table column settings

On the right upper site of a table there is a settings icon in which you can set the table columns to a desired view.
With a click on it a dialog opens where you can change the settings of the corresponding table:

image::../../images/arc42/user-guide/tx-column-selection.png[Column Settings]

Hide/show table columns by clicking on the checkbox or the column name.

> It is possible to hide/show all columns by clicking on the "All"- checkbox.

Reviewer: **Suggestion:**

```suggestion
It is possible to hide/show all columns by clicking on the "All"-checkbox.
```

The reset icon resets the table columns to its default view.

Reorder the table columns by selecting a list item (click on the right of the column name).
By selecting the column, you can reorder it with the up and down arrow icons to move it in the front or back of other columns.

Apply your changes by clicking on the "Save" - button.

> If you want to discard your changes, simply press you "ESC" - button, click anywhere else except in the dialog or close it explicitely with the close icon on the upper right of the dialog.

Reviewer: **Suggestion:**

Typo?

```suggestion
If you want to discard your changes, simply press you "ESC" - button, click anywhere else except in the dialog or close it explicitly with the close icon on the upper right of the dialog.
```

== General information

Accessing the application is done by the URL provided by the hosting company.
A list of user or technical roles can be found below:

=== Trace-X app roles

One user can have multiple roles. The available roles are:

==== User

With the user authorization, the user is able to create new alerts and investigations, view and edit existing alerts and investigations. The user can send alerts and investigations to the supervisor for approval.

==== Supervisor

With supervisor authorization, the user is able to view, evaluate and approve alerts and investigations that have been created. The supervisor also has all the authorizations that the standard user has.

==== Admin

With administrator authorization, the user is primarily able to configure the technical configuration of the EDC connection. From a technical perspective, the admin only has read authorization for existing quality alerts, investigations and parts.

== Navigation

Navigation is done based on the top menu.

image::../../images/arc42/user-guide/tx-navigation.png[Navigation]

=== Dashboard

> Provides a fast overview about the amount of manufactured (my parts) and supplier and customer parts (shared parts) and batches, as well as the amount of open investigations and alerts.

Reviewer: **Suggestion:**

Add the direction of the notifications into the documentation.

```suggestion
Provides a fast overview about the amount of manufactured (my parts) and supplier and customer parts (shared parts) and batches, as well as the amount of received open investigations and alerts.
```

=== Trace-X app roles

One user can have multiple roles. The available roles are:

==== User

With the user authorization, the user is able to create new alerts and investigations, view and edit existing alerts and investigations. The user can send alerts and investigations to the supervisor for approval.

==== Supervisor

With supervisor authorization, the user is able to view, evaluate and approve alerts and investigations that have been created. The supervisor also has all the authorizations that the standard user has.

==== Admin

> With administrator authorization, the user is primarily able to configure the technical configuration of the EDC connection. From a technical perspective, the admin only has read authorization for existing warnings, investigations and products.

Reviewer: **Clarifications:**

Where do we get `warnings` from?

=== Dashboard

> Provides a fast overview about the amount of manufactured (my parts) and supplier parts (shared parts) and batches, as well as the amount of open investigations and alerts.

Reviewer: **Suggestion:**
Shared contains both numbers, the supplier and the customer number of assets.

```suggestion
Provides a fast overview about the amount of manufactured (my parts) and supplier and customer parts (shared parts) and batches, as well as the amount of open investigations and alerts.
```

> Lists the five most recent quality investigations and alerts to get an overview of the current state.

Reviewer: Worth to mention that
- it lists the **received** notifications only.
- it lists the **active** notifications only, those in state:
  CREATED, SENT, RECEIVED, ACKNOWLEDGED, ACCEPTED, DECLINED

=== Quality alert

Navigates to the incoming and outgoing alerts.

=== Burger menu

> Use the burger menu to use view user details, access the user manual, logout or change the language.

Reviewer: **Suggestion:**

```suggestion
Use the burger menu to view user details, access the user manual, log out or change the language.
```

Reorder the table columns by selecting a list item (click on the right of the column name).
By selecting the column, you can reorder it with the up and down arrow icons to move it in the front or back of other columns.

Apply your changes by clicking on the "Save" - button.
If you want to discard your changes, simply press you "ESC" - button, click anywhere else except in the dialog or close it explicitely with the close icon on the upper right of the dialog.

The settings will be stored in the local storage of the browser and will be persisted until they get deleted.

=== Part details

Clicking on an item in the list opens "Part details" view.
More detailed information on the asset is listed as well as a part tree that visually shows the parts relations.

image::../../images/arc42/user-guide/tx-myparts-detail-relation.png[My Parts details]

==== Overview

General production information.
Information on the quality status of the part/batch.

==== Relations

Part tree based on SingleLevelBomAsBuilt aspect model.
Dependent on the semantic data model of the part the borders are in a different color.
A green border indicates that the part is a SerialPart.

> A yellow border indicates that the part is a piece of a batch.

Reviewer: **Suggestion:**

Check with UI team probably as there might be more colors, i.e. for `UNKNOWN` or `JUSTINSEQUENCE`:

A quick example:

![image](https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/assets/145972012/3d19dbbb-b2f4-4561-9bc2-6377a59fcf8e)

Additionally, it's possible to switch between tabs above each of the table views to display either supplier or customer parts

=== Supplier parts

List view of supplied parts and batches.
Supplier parts that are in a quality investigation are highlighted yellow.

==== Supplier parts select / Quality Investigation

Select one or multiple supplier parts.

> Selection will enable you to create a quality investigation (notification) to your supplier.

Reviewer: **Clarification:**

This is valid for AsBuilt only?

The quality investigation will be added to a queue (queued & requested inbox) and not directly be sent to the supplier.

Once the quality investigation is created you will get a pop-up and can directly navigate to the inbox for further action.

=== Supplier part details

Clicking on an item in the list opens "Part details" view.
More detailed information on the asset is listed.

image::../../images/arc42/user-guide/tx-sharedparts-detail.png[Shared Parts details]

==== Overview

General production information.
Information on the quality status of the supplier part/batch.

==== Manufacturer data

Detailed information on the IDs for the supplier part/batch.

==== Customer data

Information about the identifiers at the customer (in this case own company) for the respective part/batch.

=== Customer Parts

List view of customer parts and batches.

> Customer Parts that are in a quality alert are highlighted yellow.

Reviewer: **Clarification:**

Still true?

=== Customer part details

Clicking on an item in the list opens "Part details" view.
More detailed information on the asset is listed.

==== Overview

General production information.
Information on the quality status of the supplier part/batch.

==== Manufacturer data

Detailed information on the IDs for the supplier part/batch.

==== Customer data

Information about the identifiers at the customer for the respective part/batch.

== Quality investigation

In the “Quality investigation”-tab all received quality investigations are shown as well as “Queued & Requested” quality investigations which included outgoing drafted and already sent notifications.

image::../../images/arc42/user-guide/tx-qualityinvestigations.png[Quality Investigation]

> image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-drafts.png[] Received investigations.

Reviewer: **Clarification:**

Is this nice symbol available already or is it still in progress?
Can't find it in the UI.

Applies to several positions.

Received investigations by a customer specify that the given customer has found a defect or has a request for you to investigate on a specific part/ batch on your side and waits for your feedback.

image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-send.png[] Queued & Requested investigations.

Queued & Requested investigations on the other hand are investigations that were requested by your company and that are sent to the supplier after you detected a defect, or you have a request on a specific part/ batch. The given supplier is now informed that you expect feedback on the request.

> Queued investigations are investigations that were created by a user in your company but are not yet sent to the supplier until a supervisor approves them.

Reviewer: **Suggestion:**

An example screen containing a queued investigation would be helpful.

You can sort the table by clicking on the table header. On the first click, the table is sorted in ascending order, on the second click the table is sorted in descending order. If the table is already sorted and you hold shift while clicking on another table header. The table is sorted primarily by the already given column but is then ordered secondly by the specified column. You can sort by as many columns as you desire.
Next to each column header is a filter icon. If you click on that icon, a filter option menu will appear by which you can filter the table.
You can select which columns are displayed by clicking on the settings icon on the top right corner of the table.
In search bar, you can search for specific quality investigations by entering the given description or BPN (business partner number).

image::../../images/arc42/user-guide/tx-qualityinvestigation-actionmenu-highlight.png[Quality Investigation Context Menu]

image::../../images/arc42/user-guide/tx-qualityinvestigation-accept.png[Quality Investigation Dialog]

> A pop-up will notify you if the status transition was successful.

Reviewer: **Suggestion:**

Add some explanation on the images what exactly happens there? Or would it fit better in the paragraph above?

=== Quality investigation context action

Clicking on the three dots menu on the right side of a quality investigation entry opens the quality investigation context menu. In the context menu, users can open the detailed of the given quality investigation. Supervisor can additionally update the status of the given quality investigation. Only possible status update actions will be shown in the menu.

image::../../images/arc42/user-guide/tx-qualityinvestigation-actionmenu-highlight.png[Quality Investigation Context Menu]

After selecting to update the status of the quality investigation, a dialog will open in which the details of the status update are shown. The status is then updated after completing all steps in the dialog. A pop-up will notify you, it the update was successful. Reverting status updates is not intended.

image::../../images/arc42/user-guide/tx-qualityinvestigation-accept.png[Quality Investigation Dialog]

> You can sort the table by clicking on the table header. On the first click, the table is sorted in ascending order, on the second click the table is sorted in descending order. If the table is already sorted and you hold shift while clicking on another table header. The table is sorted primarily by the already given column but is then ordered secondly by the specified column. You can sort by as many columns as you desire.

Reviewer: **Suggestion:**

Use a comma before 'and' since it connects two independent clauses.

```suggestion
You can sort the table by clicking on the table header. On the first click, the table is sorted in ascending order, on the second click the table is sorted in descending order. If the table is already sorted, and you hold shift while clicking on another table header. The table is sorted primarily by the already given column but is then ordered secondly by the specified column. You can sort by as many columns as you desire.
```

Next to each column header is a filter icon. If you click on that icon, a filter option menu will appear by which you can filter the table.

> You can select which columns are displayed by clicking on the settings icon on the top right corner of the table.

Reviewer: **Suggestion:**

The usual preposition to use with "corner" is "in", not "on". Did you mean in the top right corner?

```suggestion
You can select which columns are displayed by clicking on the settings icon in the top right corner of the table.
```

image::../../images/arc42/user-guide/tx-qualityalerts.png[Quality Alerts]

image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-drafts.png[] Received alerts.

Received quality alerts by a customer specify that the given customer has found a defect or has a request for you to investigate on a specific part/ batch on your side and waits for your feedback.

image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-send.png[] Queued & Requested alerts.

Queued & Requested quality alerts on the other hand are quality alerts that were requested by your company and that are sent to the customer after you detected a defect, or you have a request on a specific part/ batch.
Queued quality alerts are quality alerts that were created by a user in your company but are not yet sent to the customer until a supervisor approves them.
Requested quality alerts are quality alerts that were approved by a supervisor in your company and consequently been sent to the customer.

=== Quality alert context action

> Clicking on the three dots menu on the right side of a quality alert entry opens the quality alert context menu. In the context menu, users can open the detailed of the given quality alert. Supervisor can additionally update the status of the given quality alert. Only possible status update actions will be shown in the menu.

Reviewer: **Suggestion:**

Typo?

```suggestion
Clicking on the three dots menu on the right side of a quality alert entry opens the quality alert context menu. In the context menu, users can open the details of the given quality alert. Supervisor can additionally update the status of the given quality alert. Only possible status update actions will be shown in the menu.
```

After selecting to update the status of the quality alert, a dialog will open in which the details of the status update are shown. The status is then updated after completing all steps in the dialog. A pop-up will notify you, it the update was successful. Reverting status updates is not intended.

> You can sort the table by clicking on the table header. On the first click, the table is sorted in ascending order, on the second click the table is sorted in descending order. If the table is already sorted and you hold shift while clicking on another table header. The table is sorted primarily by the already given column but is then ordered secondly by the specified column. You can sort by as many columns as you desire.

Reviewer: **Suggestion:**

Use a comma before 'and' since it connects two independent clauses.

```suggestion
You can sort the table by clicking on the table header. On the first click, the table is sorted in ascending order, on the second click the table is sorted in descending order. If the table is already sorted, and you hold shift while clicking on another table header. The table is sorted primarily by the already given column but is then ordered secondly by the specified column. You can sort by as many columns as you desire.
```

Next to each column header is a filter icon. If you click on that icon, a filter option menu will appear by which you can filter the table.

> You can select which columns are displayed by clicking on the settings icon on the top right corner of the table.

Reviewer: **Suggestion:**

The usual preposition to use with "corner" is "in", not "on". Did you mean in the top right corner?

```suggestion
You can select which columns are displayed by clicking on the settings icon in the top right corner of the table.
```

|===

=== Quality investigation status flow

Notifications always have a status.
The transition from one status to a subsequent status is described in the below state model.

The Sender can change the status to closed from any status.
The receiver can never change the status to closed.

The legend in the below state diagram describes who can set the status.
One exception to this rule: the transition from status SENT to status RECEIVED is done automatically once the sender receives the Http status code 201.

image::https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notificationstatemodel.png[Notification state model]

== Quality alert
In the “Quality alert”-tab all received quality alerts as well as “Queued & Requested Quality Alerts” which included outgoing drafted and already sent notifications.

image::../../images/arc42/user-guide/tx-qualityalerts.png[Quality Alerts]

image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-drafts.png[] Received alerts.

> Received quality alerts by a customer specify that the given customer has found a defect or has a request for you to investigate on a specific part/ batch on your side and waits for your feedback.

Reviewer: **Clarification:**

Isn't it more that a supplier sends quality alerts?

image:https://raw.githubusercontent.com/eclipse-tractusx/traceability-foss/main/docs/src/images/arc42/user-guide/notification-send.png[] Queued & Requested alerts.

> Queued & Requested quality alerts on the other hand are quality alerts that were requested by your company and that are sent to the supplier after you detected a defect, or you have a request on a specific part/ batch. The given supplier is now informed that you expect feedback on the request.

Reviewer: **Clarification:**

Would an alert be sent to a customer rather than a supplier?
